LTS 5.13.2 RNs.mdx (#2701)
TykTechnologies/tyk-docs: developer-support/release-notes/dashboard.mdx

## 5.13 Release Notes

### 5.13.2 Release Notes 

#### Release Date xx August 2026

#### Release Highlights

> This release fixes several Dashboard issues affecting user and API key management under load, corrects a Host header problem that broke the API Debugger behind strict service meshes like Istio, adds Dashboard-side validation to prevent APIs with an invalid Listen Path from silently failing to load on the Gateway, masks additional secrets in the Dashboard UI, and improves performance on policy updates involving many Tyk OAS APIs. This release also migrates off an unmaintained security library and updates the underlying Go toolchain to Go 1.26. This release also addresses several CVEs in dependent libraries.

Reviewer: ```suggestion
This patch implements several changes and fixes to improve the stability and correct the behavior of the Dashboard.
```

We don't need to list the bug fixes for a patch - it's not adding anything that's not in the change log.

Given the potential time difference between your upgrade and the release of this version, we recommend users verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

There are no deprecations in this release.

#### Upgrade instructions
<a id="upgrade-5.13.2"></a>

If you are upgrading to 5.13.2, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker Image to pull](https://hub.docker.com/r/tykio/tyk-dashboard/tags?page=&page_size=&ordering=&name=v5.13.2)
  - ```bash
    docker pull tykio/tyk-dashboard:v5.13.2
    ```
- Helm charts
  - [tyk-charts](/developer-support/release-notes/helm-chart)

#### Changelog
<a id="Changelog-v5.13.2" data-scroll-offset></a>

##### Changed

<AccordionGroup>
<Accordion title='Updated Golang version to 1.26'>
{/* TT-17868 */}
Tyk Dashboard has been updated to build with Go 1.26, keeping the toolchain current with upstream Go security and performance updates.
</Accordion>
</AccordionGroup>

##### Fixed

<AccordionGroup>

<Accordion title='Fixed API Debugger requests failing behind a service mesh'>
{/* TT-10248 */}

> We have resolved an issue where the Dashboard's API Debugger forwarded requests to the Gateway's `/tyk/debug` endpoint carrying the Dashboard's own Host header rather than the Gateway's. Strict routing layers such as Istio could reject these requests before they reached the Gateway. The Dashboard now sets the correct Host header when proxying debug requests, so the API Debugger works without custom service mesh workarounds.

Reviewer: ```suggestion
We have resolved an issue where requests issued by the API Debugger used the Dashboard's Host header rather than the target Gateway's. Strict routing layers such as Istio could reject these requests before they reached the Gateway. The Dashboard now sets the correct Host header when proxying debug requests, so the API Debugger works without custom service mesh workarounds.
```

</Accordion>

<Accordion title='Fixed HTTP 403 error when creating users or user groups with separate read/write database connections'>
{/* TT-15506 */}

> We have resolved an issue affecting Dashboard deployments configured with separate read and write database connection strings (SQL databases only). Creating a user or user group could return a `403` error if replication lag meant the newly written record wasn't yet visible on the read replica, even though the record had been created successfully. The Dashboard no longer performs this redundant read-after-write check, so creation succeeds reliably regardless of replication lag. This applies to user, user group, and admin user creation.

Reviewer: ```suggestion
We have resolved an issue affecting Tyk Dashboard deployments using SQL databases configured with separate read and write connection strings. Dashboard could return a `403` error if replication lag meant the newly written record wasn't yet visible on the read replica, even though the record had been created successfully. The Dashboard no longer performs a redundant read-after-write check, so creation succeeds reliably regardless of replication lag. This applies to user and user group creation.
```

<Accordion title='Added `key_hash` field to API key update responses'>
{/* TT-16433 */}
When updating an API key by its hash (`?hashed=1`), the Dashboard API previously returned the hash value in the `key_id` field of the response, rather than in `key_hash` as it does for create and get operations. The response now always includes a `key_hash` field alongside `key_id`, so key hash and key ID are consistently distinguishable across all key operations. Existing integrations that read the hash from `key_id` continue to work unchanged.
</Accordion>

<Accordion title='Fixed redundant default port in generated Tyk OAS API server URLs'>
{/* TT-16721 */}
We have resolved an issue where, if the Gateway hostname used a default port (443 for HTTPS, 80 for HTTP), the Dashboard generated a server URL that included the redundant port explicitly (for example `https://api.example.com:443` instead of `https://api.example.com`). Non-default ports (such as 8080 or 8443) continue to be included as before. New Tyk OAS APIs no longer generate this redundant default-port entry.
</Accordion>

<Accordion title='Improved performance of policy updates involving many Tyk OAS APIs'>
{/* TT-16937 */}

> We have resolved a performance issue where updating a Policy containing a large number of Tyk OAS APIs (for example, 60 or more) caused high latency, due to the Dashboard repeatedly re-parsing each API's OpenAPI document for validation, audit logging, and response serialization. Policy updates and the Gateway's `/system/policies` synchronization endpoint now scale linearly with the number of APIs in a policy instead of degrading sharply as policies grow.

Reviewer: ```suggestion
We have resolved a performance issue where updating a Policy containing a large number of Tyk OAS APIs (for example, 60 or more) caused high latency, due to the Dashboard repeatedly re-parsing each API's OpenAPI document for validation, audit logging, and response serialization. Policy updates now scale linearly with the number of APIs in a policy instead of degrading sharply as policies grow.
```

</Accordion>

<Accordion title='Masked secrets in Identity Profiles and the Dashboard API access token'>
{/* TT-17160 */}
Building on the sensitive-value masking already applied to the API Designer, the Provider Configuration Secret in Identity Profiles (the OAuth/OIDC client secret used by Social and OpenID Connect providers) and a user's Dashboard API Access Token are now masked by default in the Dashboard UI, with a reveal/hide toggle. Existing permission checks for who can view the access token are unchanged — masking only applies to users who could already see the value.
</Accordion>

<Accordion title='Dashboard now validates that Listen Path includes a leading slash'>
{/* TT-17631 */}

> We have resolved an issue where the Dashboard accepted and saved a Listen Path without a leading slash — through the API Designer, the legacy Classic API designer, and all import flows (Blueprint, Swagger, WSDL, and OpenAPI/AsyncAPI data source import). A Listen Path missing its leading slash causes the Gateway to fail to load that API, so every request to it silently returns HTTP 404. The Dashboard now validates this at both the UI and API level, and rejects the save or import with a clear error message before an invalid definition can reach the Gateway. An existing API is left unchanged if an update with an invalid Listen Path is attempted.

Reviewer: ```suggestion
We have resolved an issue where the Dashboard accepted and saved a Listen Path without a leading slash (`/`). A Listen Path missing its leading slash causes the Gateway to fail to load that API, so every request to it returns HTTP 404.

The Dashboard now validates the presence of the leading slash at both the UI and API level, and rejects the save or import with a clear error message before an invalid definition can reach the Gateway. An existing API is left unchanged if an update with an invalid Listen Path is attempted.
```

</Accordion>

> 

Reviewer: Missing TT-17730
